pkg/kernel/kernel.go

```diff
@@ -284,6 +293,7 @@ func (k *kernel) checkRunningEssentialModules() {
 }
 
 func (k *kernel) doCheckRunningEssentialModules() {
+	_ = <-k.running
```

Comment:
What is the purpose of this?

Reply:
It actually blocks until the kernel was running at some point. Otherwise you have the following scenario:

1. Module A starts
2. Module A exists, checks for running forground modules
3. Kernel still launching module B and C
4. Module A sees the kernel is not (yet) running, assumes the kernel is no longer running
5. Kernel sets running to true, but module A did already exist. You now have a kernel without any running foreground modules which did not exit

Instead, if module A waits until the kernel closes the running channel, the check correctly sees that there are no running foreground modules anymore and stops the kernel

Reply:
I understand. But this method is called inside of an infinite for loop.
So, even if this happens, it will be called some milliseconds after and the kernel will be stoped properly. Am I right?